### PR TITLE
Examples: create document for component hydration

### PR DESCRIPTION
**Pull request by samouri**
**summary**
Creates a new amp document in the `examples/` directory for showcasing component hydration.

component hydration

### CODE REVIEW

**Review by rcebulko — APPROVED** [review]

File: examples/compiler-hydration.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<!-- skip-validation -->
```
Comment (rcebulko) [inline examples/compiler-hydration.html:1]: WDYT about adding an exception to `build-system/test-configs/config.js` instead, that way you don't need to modify the validation code much?

Comment (rcebulko) [reply examples/compiler-hydration.html:1]: https://github.com/ampproject/amphtml/blob/main/build-system/test-configs/config.js#L183

Comment (samouri) [reply examples/compiler-hydration.html:1]: Potato potato

Comment (samouri) [reply examples/compiler-hydration.html:1]: Making the change now

Comment (rcebulko) [reply examples/compiler-hydration.html]: You say potato potato I say potato potato

**Pull request merged**